app/Post_Types/Archive_Item/Post_Meta.php

```diff
@@ -86,11 +86,11 @@ public function add_permanent_fields()
 
 	}
 
-	public function get_field( $post_id, $type ) {
-		$id = carbon_get_post_meta( $post_id, FieldPostMeta::FIELD_ID );
-		$label = get_the_title( $post_id );
-		$helper = carbon_get_post_meta( $post_id, FieldPostMeta::FIELD_ADMIN_HELPER_TEXT);
-		return call_user_func( array($type, 'render'), $post_id, $id, $label, $helper);
+	public function get_field( $cpt_field_id, $type ) {
+		$id = carbon_get_post_meta( $cpt_field_id, FieldPostMeta::FIELD_ID );
+		$label = get_the_title( $cpt_field_id );
+		$helper = carbon_get_post_meta( $cpt_field_id, FieldPostMeta::FIELD_ADMIN_HELPER_TEXT);
+		return call_user_func( array($type, 'render'), $cpt_field_id, $id, $label, $helper);
```

Comment:
Add a check on what is being passed here to make sure it can be called:
```php
if(is_callable(array($type,'render')){
	return call_user_func...
} else {
	return '';
}

This is a public function and as such available to third-parties too.
Leaving it as it's now is fine but the error message one would get might not be clear enough to indicate where the issue is.
```

Reply:
made some changes but not all. Will circle back if I have time